### DISCUSSION TITLE
How to get the file path from MacOS system clipboard

### DISCUSSION THREAD

**Discussion by adoyle-h** [neovim/neovim#29289] [Q&A]
## Problem

I'm using MacOS. I copy a file by pressing `⌘+c`, and want to get the filepath from system clipboard in lua.
When pressed `⌘+v` in vim insert mode and Terminal and other editors like vscode, it pasted the filepath.
But `vim.fn.getreg('*')` in nvim only return the filename, not the path.

For example, when the filepath is `/Users/adoyle/Pictures/image.png`,  `vim.fn.getreg('*')` return the value `image.png`.
But I expect to get the full filepath `/Users/adoyle/Pictures/image.png`.

How to implement it in lua?

Note:  It is a prerequisite that pressing `⌘+c` to copy file in MacOS.

## Reproduction

1. Open Finder and copy a file by pressing `⌘+c` keyboard shortcut in MacOS.
2. `nvim --clean` in terminal.
3. `:lua vim.print(vim.fn.getreg('*'))`

## Environment

I tried neovim 0.9.5 and 0.10.0, and vim 9.0. All have the same behaviors.
MacOS 14.5.

**Comment by clason** [comment]
You need to find a better clipboard provider; this is a limitation of Apple's `pbpaste` (try it on the command line!)

If you copy with `cmd-option-c`, you will get the full path, though.

**Reply by adoyle-h** [reply]
Yes, I know the `cmd-option-c`. I'm writing a plugin and want to keep it simple that user just `⌘+c` to copy file and paste the filepath in neovim.

> Note: It is a prerequisite that pressing ⌘+c to copy file in MacOS.

Is there another way to access the system clipboard in lua?

**Reply by clason** [reply]
Not on macOS.
